index.js

```diff
@@ -3,53 +3,309 @@ const state = {
       name: "carrot",
-      price: 0.35
+      price: 0.35,
+      fruit: false,
     },
     {
       id: "003-apple",
       name: "apple",
-      price: 0.35
+      price: 0.35,
+      fruit: true,
     },
     {
       id: "004-apricot",
       name: "apricot",
-      price: 0.35
+      price: 0.35,
+      fruit: true,
     },
     {
       id: "005-avocado",
       name: "avocado",
-      price: 0.35
+      price: 0.35,
+      fruit: false,
     },
     {
       id: "006-bananas",
       name: "bananas",
-      price: 0.35
+      price: 0.35,
+      fruit: true,
     },
     {
       id: "007-bell-pepper",
       name: "bell pepper",
-      price: 0.35
+      price: 0.35,
+      fruit: false,
     },
     {
       id: "008-berry",
       name: "berry",
-      price: 0.35
+      price: 0.35,
+      fruit: true,
     },
     {
       id: "009-blueberry",
       name: "blueberry",
-      price: 0.35
+      price: 0.35,
+      fruit: true,
     },
     {
       id: "010-eggplant",
       name: "eggplant",
-      price: 0.35
-    }
+      price: 0.35,
+      fruit: false,
+    },
   ],
-  cart: []
-};
\ No newline at end of file
+  cart: [],
+};
+
+// Defines constant names for the main areas
+function elementConstants() {
+  const storeList = document.querySelector(`.store--item-list`);
+  const basketList = document.querySelector(`.cart--item-list`);
+  const totalPrice = document.querySelector(`.total-number`);
+  return {
+    storeList,
+    basketList,
+    totalPrice,
+  };
+}
+
+// Renders the store
+function renderStore() {
+  const { storeList } = elementConstants();
+  storeList.innerHTML = ``
+  for (let i = 0; i < state.items.length; i++) {
+    const storeItem = createStoreItem(i);
+    storeEventListener(storeItem, i);
+    storeList.append(storeItem);
+    }
+  }
+
+// Creates a store item
+function createStoreItem(i) {
+  const currentItem = state.items[i];
+  const storeItem = document.createElement(`li`);
+    const storeItemDiv = document.createElement(`div`);
+    storeItemDiv.setAttribute(`class`, `store--item-icon`);
+      const storeItemImage = document.createElement(`img`);
+      storeItemImage.setAttribute(`src`, `assets/icons/${currentItem.id}.svg`);
+      storeItemImage.setAttribute(`alt`, currentItem.name);
+      const storeItemButton = document.createElement(`button`);
+      storeItemButton.innerText = `Add to cart`;
+  storeItemDiv.append(storeItemImage);
+  storeItem.append(storeItemDiv);
+  storeItem.append(storeItemButton);
+
+  return storeItem;
+}
+
+// Add click action to store items
+function storeEventListener(storeItem, i) {
+  const storeItemButton = storeItem.querySelector(`button`);
+  storeItemButton.addEventListener("click", () => {
+    updateBasket(i);
+    renderBasket()
+  });
+}
+
+// Updates cart array with new item or increased quantity
+function updateBasket(i) {
+  let currentItem = state.items[i];
+  if (state.cart.length === 0) {
+    state.cart.push({ ...currentItem, quantity: 1 });
```

Comment:
Could you go into this and clarify for me the part between brackets? :)

Reply:
...currentItem essentially means "everything that's in currentItem.

So if the cart length is zero, i.e. the cart is empty. then it pushes the current item to the cart array and adds a quantity property with a value of 1 to the array item.